extension_service/datastore/datastore.py

```diff
@@ -60,7 +60,20 @@ async def export_data(
         pass
 
     @abstractmethod
-    async def get_airport(self, id: int) -> Optional[models.Airport]:
+    async def get_airport_by_id(self, id: int) -> Optional[models.Airport]:
+        raise NotImplementedError("Subclass should implement this!")
+
+    @abstractmethod
+    async def get_airport_by_iata(self, iata: str) -> Optional[models.Airport]:
+        raise NotImplementedError("Subclass should implement this!")
+
+    @abstractmethod
+    async def search_airports(
+        self,
+        country: Optional[str] = None,
+        city: Optional[str] = None,
+        name: Optional[str] = None,
+    ) -> Optional[list[models.Airport]]:
```

Comment:
Removed the Optional.
It was previously Optional to keep it consistent with the Get endpoints (if no airport/amenities was found, it will return None). But I think it does make sense to return empty list for search endpoints. Updated.